src/prefect/logging/loggers.py

```diff
@@ -213,14 +224,24 @@ def print_as_log(*args, **kwargs):
         return print(*args, **kwargs)
 
     logger = get_run_logger()
+    logging.Logger.makeRecord = custom_makeRecord  # Overriding makerecord
```

Comment:
I'm very hesitant to override the `makeRecord` method on loggers like this; we'll be breaking compatibility with people who have custom loggers with their own implementation there.

Reply:
@madkinsz  Ok then I see is there any other workaround possible to fix the issue since the stack level is not optimal for all python versions